homeassistant/components/switcher_kis/light.py

```diff
@@ -35,16 +35,20 @@ async def async_setup_entry(
     def async_add_light(coordinator: SwitcherDataUpdateCoordinator) -> None:
         """Add light from Switcher device."""
         entities: list[LightEntity] = []
-        if (
-            coordinator.data.device_type.category
-            == DeviceCategory.SINGLE_SHUTTER_DUAL_LIGHT
-        ):
-            entities.extend(SwitcherDualLightEntity(coordinator, i) for i in range(2))
-        if (
-            coordinator.data.device_type.category
-            == DeviceCategory.DUAL_SHUTTER_SINGLE_LIGHT
+        number_of_lights = len(cast(SwitcherLight, coordinator.data).light)
```

Comment:
What happens if we reach this callback with a device that is not a light? this should happen after you know it is a light device

Reply:
It is done.